Resources/Computing/BatchSystems/Condor.py

```diff
@@ -152,40 +190,29 @@ def getJobStatus( self, **kwargs ):
       resultDict['Message'] = 'No user name'
       return resultDict
     
-    status,stdout_q = commands.getstatusoutput( 'condor_q -submitter %s' % user )
+    status,stdout_q = commands.getstatusoutput( 'condor_q -submitter %s -af:j JobStatus  ' % user )
     
     if status != 0:
       resultDict['Status'] = status
       resultDict['Message'] = stdout_q
       return resultDict
-      
-    status_history, stdout_history = commands.getstatusoutput( 'condor_history | grep %s' % user )  
-    
-    stdout = stdout_q
-    if status_history == 0:
-      stdout = '\n'.join( [stdout_q,stdout_history] )
+
+    qList = stdout_q.strip().split('\n')
+
+    ##FIXME: condor_history does only support j for autoformat from 8.5.3,
+    ## format adds whitespace for each field This will return a list of 1245 75 3
+    ## needs to cocatenate the first two with a dot
+    condorHistCall = 'condor_history -af ClusterId ProcId JobStatus -submitter %s' % user
+    treatCondorHistory( condorHistCall, qList )
   
     statusDict = {}
-    if len( stdout ):
-      lines = stdout.split( '\n' )
-      for line in lines:
-        l = line.strip()
-        for job in jobIDList:
-          if l.startswith( job ):
-            if " I " in line:
-              statusDict[job] = 'Waiting'
-            elif " R " in line:
-              statusDict[job] = 'Running'
-            elif " C " in line:
-              statusDict[job] = 'Done'
-            elif " X " in line:
-              statusDict[job] = 'Aborted'    
-  
-    if len( statusDict ) != len( jobIDList ):
+    if len( qList ):
       for job in jobIDList:
-        if not job in statusDict:
+        job = str(job)
+        statusDict[job] = parseCondorStatus( qList, job )
+        if statusDict[job] == 'HELD':
           statusDict[job] = 'Unknown'
```

Comment:
It did the same before. I don't use the BatchSystem/Condor, so I can't judge if this would cause problems, so I don't want to change the observed behaviour.